### DISCUSSION TITLE
couldn't run the code

### DISCUSSION THREAD

**Discussion by valtahomes** [souzatharsis/podcastfy#96] [General]
Hi 

Just install the pkg on my computer but keep receiving this error:

2024-10-23 13:38:23,750 - podcastfy.client - INFO - Processing 1 links
Error generating content: Authentication failed for [/commits/souzatharsis/podcastfy_multimodal/latest.](http://localhost:8889/commits/souzatharsis/podcastfy_multimodal/latest.) HTTPError('401 Client Error: Unauthorized for url: https://api.smith.langchain.com/commits/souzatharsis/podcastfy_multimodal/latest', '{"detail":"Using legacy API key. Please generate a new API key."}')
2024-10-23 13:38:24,274 - podcastfy.client - ERROR - An error occurred in the process_content function: Authentication failed for /commits/souzatharsis/podcastfy_multimodal/latest. HTTPError('401 Client Error: Unauthorized for url: https://api.smith.langchain.com/commits/souzatharsis/podcastfy_multimodal/latest', '{"detail":"Using legacy API key. Please generate a new API key."}')
2024-10-23 13:38:24,275 - podcastfy.client - ERROR - An error occurred: Authentication failed for /commits/souzatharsis/podcastfy_multimodal/latest. HTTPError('401 Client Error: Unauthorized for url: https://api.smith.langchain.com/commits/souzatharsis/podcastfy_multimodal/latest', '{"detail":"Using legacy API key. Please generate a new API key."}')

I did set up a new key for langsmith but still no luck.

What is the issue?

Thanks,
V

**Comment by souzatharsis** [comment]
Hi have you tried without setting up any api key for langsmith? You are not
supposed to need it... There is a warning message but you can safely ignore
it since we only use langchain but not langsmith.

**Comment by valtahomes** [comment]
Thanks. I commented out the langsmith key and it worked. But I still receive a warning as follows:

[python3.11/site-packages/langsmith/client.py:354](http://localhost:8889/Users/alex/miniconda3/envs/podcast/lib/python3.11/site-packages/langsmith/client.py#line=353): LangSmithMissingAPIKeyWarning: API key must be provided when using hosted LangSmith API
  warnings.warn(

**Comment by souzatharsis** [comment]
I'm glad it worked. Please ignore the warning. It is simply saying an api
key is required if we use langsmith but we don't.

I'll superess the warning to avoid confusion.

Thanks.

**Comment by valtahomes** [comment]
I could go through most of the code but only failed on one use case: from pdf to audio. I didn't change any of the code but it couldn't parse the pdf and I did check the pdf was there and could be opened correctly.

audio_file_from_pdf = generate_podcast(urls=["./data/pdf/s41598-024-58826-w.pdf"])

2024-10-26 10:35:48,285 - podcastfy.client - INFO - Processing 1 links
Failed to extract content from [./data/pdf/s41598-024-58826-w.pdf](http://localhost:8889/lab/tree/data/pdf/s41598-024-58826-w.pdf): URL has an invalid label.
Error extracting content from ./data/pdf/s41598-024-58826-w.pdf: Failed to extract content from ./data/pdf/s41598-024-58826-w.pdf: URL has an invalid label.
2024-10-26 10:35:48,293 - podcastfy.client - ERROR - An error occurred in the process_content function: Failed to extract content from [./data/pdf/s41598-024-58826-w.pdf](http://localhost:8889/lab/tree/data/pdf/s41598-024-58826-w.pdf): URL has an invalid label.
2024-10-26 10:35:48,293 - podcastfy.client - ERROR - An error occurred: Failed to extract content from ./data/pdf/s41598-024-58826-w.pdf: URL has an invalid label.

Do you have any idea what is wrong here?

Thanks,
V

**Comment by souzatharsis** [comment]
Please update podcastfy. This was a bug solved earlier this week.

pip install --upgrade podcastfy

**Reply by kuhnster01** [reply]
Hi there, I was experiencing the same issue even with the latest update today. I them commented out the code like this:

Raises:
------
LangSmithUserError
    If the API key is not provided when using the hosted service.
"""
# If the domain is langchain.com, raise error if no api_key
"""if not api_key:
    if _is_langchain_hosted(api_url):
        warnings.warn(
            "API key must be provided when using hosted LangSmith API",
            ls_utils.LangSmithMissingAPIKeyWarning,
        )"""
But now, when running: python -m podcastfy.client --url https://www.mysite.com/

I get this error ...

Retrying langchain_google_genai.chat_models._chat_with_retry.._chat_with_retry in 2.0 seconds as it raised RetryError: Timeout of 600.0s exceeded, last exception: 503 Getting metadata from plugin failed with error: ('invalid_grant: Bad Request', {'error': 'invalid_grant', 'error_description': 'Bad Request'}).

**Comment by kuhnster01** [comment]
Hi there, I was experiencing the same issue even with the latest update today.  I them commented out the code like this:

    Raises:
    ------
    LangSmithUserError
        If the API key is not provided when using the hosted service.
    """
    # If the domain is langchain.com, raise error if no api_key
    """if not api_key:
        if _is_langchain_hosted(api_url):
            warnings.warn(
                "API key must be provided when using hosted LangSmith API",
                ls_utils.LangSmithMissingAPIKeyWarning,
            )"""

But now, when running: python -m podcastfy.client --url https://www.mysite.com

I get this error ... 

Retrying langchain_google_genai.chat_models._chat_with_retry.<locals>._chat_with_retry in 2.0 seconds as it raised RetryError: Timeout of 600.0s exceeded, last exception: 503 Getting metadata from plugin failed with error: ('invalid_grant: Bad Request', {'error': 'invalid_grant', 'error_description': 'Bad Request'}).

**Reply by kuhnster01** [reply]
I resolved my issue. Please ignore the above. All is working

**Comment by souzatharsis** [comment]
Thanks for confirming.
